src/java.base/share/classes/java/util/Random.java

```diff
@@ -80,8 +81,143 @@
         name = "Random",
         i = 48, j = 0, k = 0,
         equidistribution = 0
-)
+        )
 public class Random implements RandomGenerator, java.io.Serializable {
+
+    /**
+     * Class used to wrap a {@link java.util.random.RandomGenerator} to
+     * {@link java.util.Random}.
+     */
+
+    @SuppressWarnings("serial")
+    private static class RandomWrapper extends Random implements RandomGenerator {
+        private final RandomGenerator generator;
+        private final boolean initialized;
```

Comment:
Can we create a private or package-private constructor for `Random` for this subclass' constructor to call? The no-arg constructor has some unnecessary calculations, and the `long` constructor calls `setSeed`. A special `Random` constructor for this subclass allows removing this special logic (by not calling `setSeed` at all) and avoid redundant seed initialization.

Reply:
More specifically, in Random class, declare
```java
// special constructor for RandomWrapper to call
private Random(Void unused) {
    this.seed = new AtomicLong();
}
```

So the randomwrapper can be (with `initialized` field removed):
```java
private RandomWrapper(RandomGenerator randomToWrap) {
    super(null);
    this.generator = randomToWrap;
}

@Override
public void setSeed(long seed) {
    throw new UnsupportedOperationException();
}
```

Currently, the RandomWrapper constructor calls the no-arg super constructor implicitly, which has some overheads for unnecessary calculations.